test/setup.ts

```diff
@@ -0,0 +1,50 @@
+import 'core-js/proposals/reflect-metadata'
+import { NydusClient, NydusServer } from 'nydus'
+import { container, instanceCachingFactory } from 'tsyringe'
+import { UpdateOrInsertUserIp } from '../server/lib/network/user-ips-type'
+import { RequestSessionLookup, SessionInfo } from '../server/lib/websockets/session-lookup'
+import {
+  ClientSocketsGroup,
+  ClientSocketsManager,
+  UserSocketsGroup,
+  UserSocketsManager,
+} from '../server/lib/websockets/socket-groups'
+
+export class FakeNydusServer extends NydusServer {
+  publish() {
+    console.log('FakeNydusServer - publish')
+  }
+}
+
+export class FakeUserSocketsManager extends UserSocketsManager {
```

Comment:
Ideally these don't need to be faked, but the goal here should probably be that the state of all these things (NydusServer, UserSocketsManager, ClientSocketsManager) aligns to the same thing without manual effort. That's why I'd recommend faking the session lookup instead. May help to make a utility method somewhere to create/connect fake clients with a particular username/client ID.